src/rust/ide/span-tree/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+//! SpanTree module
+//!
+//! SpanTree is astructure describing expression with nodes mapped to expression text spans. It can
+//! be be considered a layer over AST, that add an information about chains (you can
+//! iterate over all elements of infix chain like `1 + 2 + 3` or prefix chain like `foo bar baz`),
+//! and provides interface for AST operations like set node to a new AST or add new element to
+//! operator chain.
+
+#![feature(associated_type_bounds)]
+#![feature(trait_alias)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+pub mod action;
+pub mod generate;
+pub mod iter;
+pub mod node;
+#[cfg(test)]
+pub mod builder;
+
+pub use node::Node;
+
+pub mod traits {
+    pub use crate::action::SpanTreeActions;
+    pub use crate::generate::SpanTreeGenerator;
+    #[cfg(test)]
+    pub use crate::builder::Builder;
+}
+
+/// Common types that should be visible across the whole crate.
+pub mod prelude {
+    pub use crate::traits::*;
+    pub use enso_prelude::*;
+    pub use utils::fail::FallibleResult;
+}
+
+
+use prelude::*;
+
+// ================
+// === SpanTree ===
+// ================
+
+/// A SpanTree main structure.
+///
+/// This structure is used to have some specific node marked as root node, to avoid confusion
+/// regarding SpanTree crumbs and AST crumbs.
+#[derive(Debug,Eq,PartialEq)]
+pub struct SpanTree {
+    /// A root node of the tree.
+    pub root : Node
+}
+
+impl SpanTree {
```

Comment:
I'd be happy to have some `new` function or similar here. That'd help with discoverability, as this is the first thing I try when trying to figure out how to obtain value of given type.